### DISCUSSION TITLE
Is it possible to have more than one "blog-start.html" file?

### DISCUSSION THREAD

**Discussion by PetersOtto** [datenstrom/community#849] [Ask a question] [answered]
Hello!

I am in the process of building an events page.
For this I have installed two blogs, as it is described in the post [#108](https://github.com/datenstrom/yellow/discussions/108#issuecomment-266282194).

Now I want to create a separate `blog-start.html` for each blog.

I duplicate the `blog-start.html` and name one of it to ` news-start.html`. The header of the `page.md` looks like this:

```
---
Title: news
Layout: news-start
LayoutNew: blog
---
```
But this does not work. What am I doing wrong?

**Comment by IngoBelka** [comment]
As far as I find in blog.php, the use of layout "blog-start" is hard coded there. So one have to make blog.php using a variable layout, or you need a copy of blog extension, which uses a different layout. If I'm understandig the source code well?!

**Reply by PetersOtto** [reply]
Hello Ingo,
it seems not as easy as I hoped :)

However, with Yellow there are often more than one solutions. I have been able to solve my problem with installing only one blog.

**Reply by PetersOtto** [reply]
This is how I solved the problem. Maybe it will help someone &#128515;
One blog, two layouts.

1. Fresh Yellow install with the little blog.

2. Create folder `3-news`

3. Create `page.md` inside this folder with the following header:
```
---
Title: News
Layout: news-start
LayoutNew: news
---
```

4. Create `page-new-news.md` in the folder `shared` and insert the following:
```
---
Title: Newsseite
Published: @datetime
Author: @username
Layout: news
Cat: news
---

Dies ist eine neue Newsseite. 
```

5. Copy the layout file `blog-start.html`, rename it to `news-start.html` and replace

`<?php $pages = $this->yellow->page->getPages("blog")->paginate($this->yellow->system->get("blogPaginationLimit")) ?>`

to

`<?php $pages = $this->yellow->content->index()->filter("cat", "news")->sort("published", false)->paginate(10) ?>`

6. Copy the layout file `blog.html` and rename it to `news.html`.

7. Now you can style the news site different to the blog site.

**Reply by PetersOtto** [reply]
If you want to use filter pages for e.g. tag or author, then you have to insert some more code in the `news-start.html.` 
It is about point 5. in the instructions above.

5. Copy the layout file `blog-start.html`, rename it to `news-start.html` and replace

`<?php $pages = $this->yellow->page->getPages("blog")->paginate($this->yellow->system->get("blogPaginationLimit")) ?>`

to

```
<?php $pages = $this->yellow->content->index()->filter("cat", "news")->sort("published", false)->paginate(10) ?>
<?php foreach ($pages as $page): ?>

<?php if ($page->isRequest("tag")): ?>
<?php $tag = $page->getRequestHtml("tag") ?>
<?php $pages = $this->yellow->content->index()->filter("cat", "news")->filter("tag", $tag)->sort("published", false)->paginate(10) ?>
<?php endif ?>

<?php if ($page->isRequest("author")): ?>
<?php $author = $page->getRequestHtml("author") ?>
<?php $pages = $this->yellow->content->index()->filter("cat", "news")->filter("author", $author)->sort("published", false)->paginate(10) ?>
<?php endif ?>
<?php endforeach ?>
```

Note: https://github.com/datenstrom/yellow/discussions/933#discussion-6134894

**Comment by annaesvensson** [comment, marked as answer]
No, this is currently not possible with the [blog extension](https://github.com/annaesvensson/yellow-blog). The HTML output for the blog start page is generated by one layout file. If you want different HTML output on different blog start pages, you can add a page setting and handle the condition in your layout file. Here's an example page setting `ShowCustomType`:

~~~
---
Title: News
Layout: blog-start
LayoutNew: blog
ShowCustomType: 1
---
~~~

Change the following in your layout file `system/layouts/blog-start.html`:
~~~
<?php if ($this->yellow->page->get("showCustomType")): ?>
...
<?php else: ?>
...
<?php endif ?>
~~~

As an alternative use a different extension. See also datenstrom/community#417.

**Reply by PetersOtto** [reply]
Thank you very much! I will try it out.

**Reply by IngoBelka** [reply]
@annaesvensson - This is a pretty cool solution, thank you for this suggestion!
